File: dcp-solutions/2024-09-27/solution.py
```python
def maxSubarrayLength(nums, k):
    if not nums:
        return []

    max_sum = [0] * len(nums)
    max_sum[0] = nums[0]

    for i in range(1, k):
        max_sum[i] = max(max_sum[i-1], nums[i])

    for i in range(k, len(nums)):
        max_sum[i-k+1] = max(max_sum[i-k], sum(nums[i-k+1:i+1]))
        
        print(max_sum[i], end=" ")

# Test the function
maxSubarrayLength([10, 5, 2, 7, 8, 7], 3)
```

File: dcp-solutions/2024-09-27/solution.js
```javascript
function maxSubarrayLength(nums, k) {
    const n = nums.length;
    let windowSum = 0;
    let maxNums = new Array(n).fill(0);
    let left = 0;

    for (let right = 0; right < n; right++) {
        windowSum += nums[right];
        
        if (right >= k - 1) {
            maxNums[left] = Math.max(maxNums[left], windowSum);
            windowSum -= nums[left];
            left++;
        }
    }

    for (let i = k; i < n; i++) {
        process.stdout.write(maxNums[i - k] + " ");
    }
}

// test the function
maxSubarrayLength([10, 5, 2, 7, 8], 3);
```